lib/user_lookup.rb

```diff
@@ -1,9 +1,16 @@
 # frozen_string_literal: true
 
 class UserLookup
+  def self.lookup_columns
+    @user_lookup_columns ||= %i{id username name uploaded_avatar_id primary_group_id flair_group_id admin moderator trust_level}
+  end
+
+  def self.group_lookup_columns
+    @group_lookup_columns ||= %i{id name flair_icon flair_upload_id flair_bg_color flair_color}
+  end
 
   def initialize(user_ids = [])
-    @user_ids = user_ids.tap(&:compact!).tap(&:uniq!).tap(&:flatten!)
+    @user_ids = user_ids.compact.uniq.flatten
```

Comment:
Note that this will allocate 3 extra arrays as compared to the previous implementation.

Reply:
Imo the difference in memory characteristics is negligible. (and `#flatten!` does create a new array anyway) Readability and consistency takes a priority here (as it does in most cases)

Reply:
It depends on the size of user_ids in this case. If the number of elements in the array is huge, the extra memory is non-negligible.